draft/XXXX-table-metadata/README.md

```diff
@@ -0,0 +1,84 @@
+
+## Proposals
+
+#### Data Model
+There are several way we could go about this, but the cleanest way is to store this information in a special collection, in the workspace that the table resides in. This special collection would be named something like `multinet_meta`, and each entry would correspond to the metadata on a specific graph or table. The data model would look something like this:
+
+
+```
+{
+  item_id: str                 // The ID of the table/graph this data belongs to
+  item_type: "table" | "graph" // The major type of the item
+  table?: {                    // Only present if the major type is "table"
+    type: "edge" | "node"      // The table type
+    columns: [
+      {
+        key: str               // E.g. {"foo": "bar"}, "foo" is the key
+        type: "number" | "label" | "category" | "date" | "bool"
+      }
+    ]
+  }
+  graph?: {                    // Only present if the major type is "graph"
+    // TBD
+  }
+
+}
+```
+
+#### API
+There are two main operations that are necessary for this functionality.
+
+1. Specifying the metadata on table creation
+2. Updating this metadata
+3. Retrieving this metadata
+
+Ideally, our API would allow for the creation of a table/graph record, with more versatile options for specifying metadata. However, our API isn't currently structured this way. Instead, in order to conform to our existing API, we would likely just add this metadata object as a query parameter onto each upload endpoint. For retrieving and updating this metadata, the API should resemble the following:
+
+```
+GET /api/workspaces/{workspace}/tables/{table}/metadata
+PUT /api/workspaces/{workspace}/tables/{table}/metadata
+```
+
+**Note:** Due to the restrictions of our current API, the endpoint used create a table/graph is also used to initially pass along the table/graph metadata, while the `PUT` endpoint above is used for updating this metadata. In the future, this `PUT` action can be used for both initially specifying metadata, and for updating this metadata (that is, as far as the API is concerned, they are the same action).
+
+##### Specifying Metadata on Creation
+
+The data model of the API would differ slightly from that of what we store internally, to allow for a single request to describe metadata on multiple entities (mainly in the case of graphs). The data format of this API may resemble the following:
+
+```
+{
+  graph?: {
+    // TBD, any relevant graph metadata
+  }
+  tables: [  // For a single CSV, this array must contain only one item.
+    {
+      // Unchanged from the table field above
+    }
+  ]
+}
+```
+
+However, handling the case of graphs may not be necessary in the first approach. Feedback on how to best implement this is welcome.
```

Comment:
Yeah, let's not worry about network metadata in the current iteration. If you want, you can just punt on that in this text:

```suggestion
In this iteration, network metadata will not be implemented.
```